### DISCUSSION TITLE
Template StringReplace for auto-generated values (e.g. tp.file.title)

### DISCUSSION THREAD

**Discussion by Gewerd-Strauss** [SilentVoid13/Templater#534] [Help] [answered]
Hello,

For my daily notes, I use a simple template for navigating and auto-correlation:

    #<% tp.file.title %>
    << [[<% tp.date.now("DD.MM.YYYY",-1,tp.file.title,"DD.MM.YYYY") %>]] || [[<% tp.date.now("DD.MM.YYYY",1,tp.file.title,"DD.MM.YYYY") %>]] >>
    bunch of other stuff, not necessary for this
    

Files are created via the Calendar-plugin, residing in `DailyNotes/YYYY/MM MMMM/DD.MM.YYYY.md`. So we get Year-folders that are purely numeric (`2022`), monthly subfolders which autosort (`01 January`, `02 February`), and finally the file-name themselves (`25.01.2022.md`)

Now, I am currently using the `#<% tp.file.title %>`-portion to create a partial tag `#01.02.2022`. I then manually convert that to `#2022-02`. And in principle I could do so with a  custom Autohotkey-script, mostly using RegEx and a bit of logic to get the right position myself. But... that seems like a bit overkill, because I believe it can be achieved in JS as well.

---

Now, I ask if it is possible to get this step done in templater itself. I unfortunately also have no knowledge about coding in JS, so I am a bit at a loss as to how one would even go about setting up custom functions. I know it's possible as per the plugin's description, but I lack even the barest knowledge for JS. 

---

Thank you,  
Sincerely,  
~Gw

**Comment by I-Pch** [comment, marked as answer]
Hi @Gewerd-Strauss 😊 

If I understand correctly, you want to display in your daily note a `#tag` formatted as `#YYYY-MM` (at the creation of the daily note)? 😇 

If so, this can be done simply with this small template : 
```javascript
<% tp.date.now("#YYYY-MM",0,tp.file.title,"DD.MM.YYYY") %>
```
or this one:
```javascript
<% moment(tp.file.title,'DD.MM.YYYY').format('#YYYY-MM') %>
```
Like for the navigation you already use they rely on the title of your daily notes and how they are formatted 😊 

Now, I'm sorry in advance if I misunderstood your question and this is not what you were looking for 😅...

**Reply by Gewerd-Strauss** [reply]
Hui, that worked flawlessly. 

Thank you :)

**Reply by I-Pch** [reply]
No problem @Gewerd-Strauss 😁 !
I'm glad to know it worked 🎉 !
